server/server/data_loader/arangoload.py

```diff
@@ -250,16 +355,38 @@ def add_root_docs_and_edges(change_tracker, db, structure_dir):
     name_docs = []
     edges = []
     mapping = {}
+
+    tree_dir = structure_dir / 'tree'
+
     division_files = sorted((structure_dir / 'division').glob('**/*.json'))
+    names_files = sorted((structure_dir / 'name').glob('**/*.json'))
+    tree_files = sorted(tree_dir.glob('**/*.json'))
     category_files = sorted(structure_dir.glob('*.json'))
 
-    root_languages = process_root_languages(structure_dir)
+    super_tree_file = tree_dir / 'super-tree.json'
+    tree_files.remove(super_tree_file)
+
+    root_languages = process_root_languages(structure_dir / 'super_root_lang.json')
+    super_extra_info = process_extra_info_file(structure_dir / 'super_extra_info.json')
+    text_extra_info = process_extra_info_file(structure_dir / 'text_extra_info.json')
 
     if change_tracker.is_any_file_new_or_changed(
         division_files + category_files
     ) or change_tracker.is_any_function_changed(
-        [process_division_files, process_category_files, perform_update_queries]
+        [process_division_files, process_category_files, perform_update_queries,
+         process_super_tree_file, process_tree_files, process_names_files, parse_tree_recursive]
     ):
+        nav_details_docs = process_names_files(
+            names_files,
+            root_languages,
+            super_extra_info,
+            text_extra_info
+        )
+
+        super_tree_edges = process_super_tree_file(super_tree_file)
+        nav_details_edges = process_tree_files(tree_files)
+        nav_details_edges.extend(super_tree_edges)
```

Comment:
I'd probably do `process_tree_files(super_tree_file, tree_files)`, just to keep the logic of generating nav_details in one place, but this one is fine too.

Reply:
actually, I was thinking about it, next time I will create a separate function, thanks!